q2_diversity/_alpha/_visualizer.py

```diff
@@ -193,3 +200,152 @@ def alpha_correlation(output_dir: str,
+
+    if max_depth <= min_depth:
+        raise ValueError('Provided max_depth of %d must be greater than '
+                         'provided min_depth of %d.' % (max_depth, min_depth))
+    max_frequency = max(feature_table.sum(axis='sample'))
+    if max_frequency < max_depth:
+        raise ValueError('Provided max_depth of %d is greater than '
+                         'the maximum sample total frequency of the '
+                         'feature_table (%d).' % (max_depth, max_frequency))
+    filenames, categories = [], []
+    data = _compute_rarefaction_data(feature_table, min_depth, max_depth,
+                                     steps, iterations, phylogeny, metrics)
+    for (m, data) in data.items():
+        metric_name = quote(m)
+        filename = '%s.csv' % metric_name
+
+        if metadata is None:
+            n_df = _compute_summary(data, 'sample-id')
+            jsonp_filename = '%s.jsonp' % metric_name
+            _beta_rarefaction_jsonp(output_dir, jsonp_filename, metric_name,
+                                    n_df, '')
+            filenames.append(jsonp_filename)
+        else:
+            metadata_df = metadata.to_dataframe()
+            metadata_df.columns = pd.MultiIndex.from_tuples(
+                [(c, '') for c in metadata_df.columns])
+            merged = data.join(metadata_df, how='left')
+            categories = metadata_df.columns.get_level_values(0)
+            for category in categories:
+                category_name = quote(category)
+                reindexed_df, counts = _reindex_with_metadata(category,
+                                                              categories,
+                                                              merged)
+                c_df = _compute_summary(reindexed_df, category, counts=counts)
+                jsonp_filename = "%s-%s.jsonp" % (metric_name, category_name)
+                _beta_rarefaction_jsonp(output_dir, jsonp_filename,
+                                        metric_name, c_df, category_name)
+                filenames.append(jsonp_filename)
+
+        with open(os.path.join(output_dir, filename), 'w') as fh:
+            data.columns = ['depth-%d_iter-%d' % (t[0], t[1])
+                            for t in data.columns.values]
+            if metadata is not None:
+                data = data.join(metadata.to_dataframe(), how='left')
+            data.to_csv(fh, index_label=['sample-id'])
+
+    index = os.path.join(TEMPLATES, 'alpha_rarefaction_assets', 'index.html')
+    q2templates.render(index, output_dir,
+                       context={'metrics': list(metrics),
+                                'filenames': filenames,
+                                'categories': list(categories)})
+
+    shutil.copytree(os.path.join(TEMPLATES, 'alpha_rarefaction_assets',
+                                 'dist'),
+                    os.path.join(output_dir, 'dist'))
+
+
+alpha_rarefaction_supported_methods = (non_phylogenetic_metrics()
```

Comment:
This will always be empty because set intersection is being used instead of a union:

```python
In [3]: alpha_rarefaction_supported_methods
Out[3]: set()
```

Note: there's [a bug in the framework](https://github.com/qiime2/qiime2/issues/321) where `Str % Choices` with an empty set of choices degrades to `Str` instead of raising an error.